src/ripple/core/impl/LoadEvent.cpp

```diff
@@ -29,73 +30,74 @@ LoadEvent::LoadEvent (
         LoadMonitor& monitor,
         std::string const& name,
         bool shouldStart)
-    : m_loadMonitor (monitor)
-    , m_isRunning (shouldStart)
-    , m_name (name)
-    , interval_ { std::chrono::steady_clock::now() }
+    : monitor_ (monitor)
+    , running_ (shouldStart)
+    , name_ (name)
+    , mark_ { std::chrono::steady_clock::now() }
     , timeWaiting_ {}
     , timeRunning_ {}
 {
 }
 
 LoadEvent::~LoadEvent ()
 {
-    if (m_isRunning)
+    if (running_)
         stop ();
 }
 
 std::string const& LoadEvent::name () const
 {
-    return m_name;
+    return name_;
 }
 
 std::chrono::milliseconds
-LoadEvent::getSecondsWaiting() const
+LoadEvent::waitTime() const
 {
-    return timeWaiting_;
+    // C++17: std::chrono::ceil
+    return ceil<std::chrono::milliseconds>(timeWaiting_);
```

Comment:
I recommend your second option:  report `std::chrono::steady_clock::duration`.  That simplifies things by having the rounding in one place (`LoadMonitor`).  All this stuff with different precisions in the old code seems to be an artifact of not using a library such as `chrono` which makes multi-precision comparisons painless.

The results of `getSecondsRunning` and `getSecondsWaiting` are used only for logging, and so it really doesn't matter much how they are rounded or formatted:  Whatever looks best to the human reader.

The use of `getSecondsTotal` can easily deal with `chrono::nanoseconds` (`std::chrono::steady_clock::duration`) and whether or not to replicate the current behavior (map durations <= 1ms to 0ms) is a separable question that @JoelKatz is perhaps better qualified to address.